tests/test_all_gather.py

```diff
@@ -0,0 +1,38 @@
+# Copyright (c) 2023 Graphcore Ltd. All rights reserved.
+
+import torch
+import poptorch
+import poptorch_experimental_addons as pea
+import random
+import numpy as np
+import einops
```

Comment:
CI will complain about sort-imports. On vscode, I haven't managed to get it to run this automatically as well as black, but just type `Ctrl+Shift+P Sort...`.